tests/test_inference.py

```diff
@@ -1667,6 +1639,36 @@ def test_oldest_ancestors(self):
         assert np.isclose(ultimate_root_time - root_time, expected_time_diff)
         assert np.isclose(root_time - oldest_non_root, expected_time_diff)
 
+    @pytest.mark.parametrize("genotype_encoding", tsinfer.GenotypeEncoding)
+    @pytest.mark.parametrize("num_threads", [0, 1, 10])
+    def test_encodings_identical_results(self, genotype_encoding, num_threads):
+        n = 26
+        m = 57
+        G, positions = get_random_data_example(n, m)
+        sample_data = tsinfer.SampleData(sequence_length=m)
+        for genotypes, position in zip(G, positions):
+            sample_data.add_site(position, genotypes)
+        sample_data.finalise()
+        # Do we get precisely the same result as the Python engine with no encoding?
+        a1 = tsinfer.generate_ancestors(sample_data, engine=tsinfer.PY_ENGINE)
+        a2 = tsinfer.generate_ancestors(
+            sample_data, genotype_encoding=genotype_encoding, num_threads=num_threads
+        )
+        a1.assert_data_equal(a2)
+
+    def test_one_bit_encoding_missing_data(self):
```

Comment:
Do we also need to test the case of no missingness but 3 states?

Reply:
Good idea, but can't test because mulilti allelic don't get through to ancestor gen